### PR TITLE
ValuesSource refactoring: Wire up Sum aggregation

### PR DESCRIPTION
**Pull request by csoulios**
Wire up Sum aggregation for #42949

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/metrics/SumAggregatorFactory.java [hunk 1]
```diff
@@ -46,6 +50,22 @@
         super(name, config, queryShardContext, parent, subFactoriesBuilder, metaData);
     }
 
+    static void registerAggregators(ValuesSourceRegistry valuesSourceRegistry) {
+        valuesSourceRegistry.register(SumAggregationBuilder.NAME,
+           List.of(CoreValuesSourceType.NUMERIC, CoreValuesSourceType.DATE),
```
Comment (not-napoleon) [inline server/src/main/java/org/elasticsearch/search/aggregations/metrics/SumAggregatorFactory.java]: Should we support booleans here?  I haven't checked Sum specifically, but I'd bet it accepts a boolean field in master.

**Review by not-napoleon — APPROVED** [review]
Left one comment that needs to be addressed, but otherwise looks good to me.

File: server/src/main/java/org/elasticsearch/search/aggregations/metrics/SumAggregatorFactory.java (hunk 1 above)
Comment (csoulios) [reply server/src/main/java/org/elasticsearch/search/aggregations/metrics/SumAggregatorFactory.java]: You're right. I am adding booleans too.

**Comment by csoulios** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix-unix

**Pull request merged**